src/apscheduler/datastores/asyncpg.py

```diff
@@ -0,0 +1,582 @@
+from __future__ import annotations
+
+import sys
+from collections import defaultdict
+from contextlib import AsyncExitStack
+from datetime import datetime, timedelta, timezone
+from pathlib import Path
+from typing import Any, Iterable
+from uuid import UUID
+
+import anyio
+import attrs
+import sniffio
+import tenacity as tenacity
+from asyncpg import Connection, Pool, create_pool
+from asyncpg.protocol.protocol import Record
+
+from .. import (
+    ConflictPolicy,
+    DataStoreEvent,
+    Job,
+    JobAcquired,
+    JobAdded,
+    JobDeserializationFailed,
+    JobResult,
+    Schedule,
+    ScheduleAdded,
+    ScheduleDeserializationFailed,
+    ScheduleRemoved,
+    ScheduleUpdated,
+    SerializationError,
+    Task,
+    TaskAdded,
+    TaskLookupError,
+    TaskRemoved,
+    TaskUpdated,
+)
+from ..abc import EventBroker
+from ..marshalling import callable_to_ref
+from .base import BaseExternalDataStore
+
+if sys.version_info >= (3, 11):
+    from typing import Self
+else:
+    from typing_extensions import Self
+
+
+class MyRecord(Record):
+    def __getattr__(self, name):
+        return self[name]
+
+
+@attrs.define(eq=False)
+class AsyncPgDataStore(BaseExternalDataStore):
+    dsn: str
+    pool: Pool | None = attrs.field(default=None)
+    schema: str | None = attrs.field(default=None)
+    _tables: list[str] = attrs.field(
+        default=["metadata", "tasks", "schedules", "jobs", "job_results"]
+    )
+    _dbtypes: list[str] = ["coalescepolicy", "joboutcome"]
+
+    @classmethod
+    async def from_dsn(
+        cls, dsn: str, schema: str, start_from_scratch: bool, **pool_options
+    ) -> Self:
+        self = AsyncPgDataStore(
+            dsn=dsn,
+            schema=schema,
+            start_from_scratch=start_from_scratch,
+            **pool_options,
+        )
+        self.pool = await create_pool(
+            dsn=dsn, server_settings={"search_path": schema}, **pool_options
```

Comment:
this avoid having to use the whole dance about prepending table names with the schema name you have to do in the sqlalchemy datastore, not sure it is adaptable though